subprojects/code-quality/src/testFixtures/groovy/org/gradle/quality/integtest/fixtures/CodeNarcCoverage.groovy

```diff
@@ -18,14 +18,20 @@ package org.gradle.quality.integtest.fixtures
 
 import org.gradle.api.JavaVersion
 import org.gradle.api.plugins.quality.CodeNarcPlugin
+import org.gradle.util.TestPrecondition
 import org.gradle.util.VersionNumber
 
 class CodeNarcCoverage {
     private final static List<String> ALL = ["0.17", "0.21", "0.23", "0.24.1", "0.25.2", "1.0", "1.2.1", CodeNarcPlugin.DEFAULT_CODENARC_VERSION].asImmutable()
 
     private final static List<String> JDK11_SUPPORTED = ALL.findAll { VersionNumber.parse(it) > VersionNumber.parse("0.21") }.asImmutable()
+    // TODO: document this
+    private final static List<String> JDK14_SUPPORTED = ALL - ['0.17', '0.21', '1.0', '1.2.1', '1.5']
```

Comment:
Is this the right list?  JDK11 says it only supports 0.21+.  Is this based more on the version of Groovy being used?

Reply:
If this comes down to just the version of Groovy that's being used and not the version of CodeNarc, I think we should test what the user is going to try:

1. If we can just rely on whatever version of Groovy that CodeNarc suggests, we use that.
2. For JDKs that don't work with that version of Groovy, try to use the minimum version of Groovy that supports that JDK.

Reply:
Forcing a Groovy version for 1.5 does the trick: https://github.com/gradle/gradle/pull/12409